lux/test/unit/lux/prisms/messages_test.exs

```diff
@@ -0,0 +1,238 @@
+defmodule Lux.Prisms.Discord.MessagesTest do
+  use ExUnit.Case
+  import Mock
+
+  alias Lux.Prisms.Discord.Messages.{
+    SendMessagePrism,
+    EditMessagePrism,
+    DeleteMessagePrism,
+    PinMessagePrism,
+    UnpinMessagePrism,
+    GetPinnedMessagesPrism,
+    BulkDeleteMessagesPrism,
+    ReactToMessagePrism
+  }
+  alias Lux.Lenses.DiscordLens
+
+  @valid_channel_id "123456789012345678"
+  @valid_message_id "876543210987654321"
+  @agent_ctx %{agent: %{name: "TestAgent"}}
+
+  describe "SendMessagePrism" do
+    test "successfully sends a message" do
+      message_content = "Test message"
+      response = %{
+        "id" => @valid_message_id,
+        "content" => message_content,
+        "channel_id" => @valid_channel_id
+      }
+
+      with_mock DiscordLens, [
+        focus: fn %{
+          endpoint: "/channels/" <> _,
+          method: :post,
+          body: %{content: ^message_content}
+        } -> {:ok, response} end
+      ] do
+        assert {:ok, %{message: ^response}} =
+          SendMessagePrism.handler(
+            %{
+              channel_id: @valid_channel_id,
+              content: message_content
+            },
+            @agent_ctx
+          )
+      end
+    end
+  end
+
+  describe "EditMessagePrism" do
+    test "successfully edits a message" do
+      new_content = "Updated message"
+      response = %{
+        "id" => @valid_message_id,
+        "content" => new_content,
+        "edited_timestamp" => "2024-01-01T00:00:00.000Z"
+      }
+
+      with_mock DiscordLens, [
```

Comment:
Please take a look at Req.Test and if you do the Discord.Client, you could mock that instead. I think it's better to mock Req directly so that you can actually test your code after the HTTP requests are done.